Release/include/cpprest/http_client.h

```diff
@@ -104,7 +104,7 @@ class http_client_config
 #if !defined(_WIN32) && !defined(__cplusplus_winrt) || defined(CPPREST_FORCE_HTTP_CLIENT_ASIO)
         , m_tlsext_sni_enabled(true)
 #endif
-#if defined(_WIN32) && !defined(__cplusplus_winrt)
+#if (defined(_WIN32) && !defined(__cplusplus_winrt)) || defined(CPPREST_FORCE_HTTP_CLIENT_CURL)
```

Comment:
Hmmm I think with this we are moving to a place where the backend is not just a function of the platform, it's something someone might want to trigger from. We should probably change the build infrastructure to set CPPREST_FORCE_HTTP_CLIENT_Xxx where Xxx can be one of the backends, and remove all of this repeated platform testing (here `(defined(_WIN32) && !defined(__cplusplus_winrt))`) in the codebase.

That's not directly related to this review of course but I want to investigate doing that before merging this.